### PR TITLE
Added Precision & Recall Metrics with average and threshold parameter

### PR DESCRIPTION
**Pull request by minesh1291**
# Description

Brief Description of the PR:

Fixes # (issue)

## Type of change

- [x] New Metric and the changes conform to the [metric contribution guidelines](https://github.com/tensorflow/addons/blob/master/tensorflow_addons/metrics/README.md#contribution-guidelines)

# Checklist:

- [x] I've properly [formatted my code according to the guidelines](https://github.com/tensorflow/addons/blob/master/CONTRIBUTING.md#coding-style)
    - [x] By running Black + Flake8
    - [x] By running pre-commit hooks
- [x] I have made corresponding changes to the documentation

# How Has This Been Tested?

If you're adding a bugfix or new feature please describe the tests that you ran to verify your changes:
*

### CODE REVIEW

**Comment by minesh1291** [comment]
do I have to take any action to push this forward!

**Comment by bhack** [comment]
I think this component is orphan. Are you interested to be a codeowner?

**Comment by minesh1291** [comment]
Yes, I will be interested.

**Comment by bhack** [comment]
@minesh1291 Before we change the codeowner can you open a ticket in https://github.com/keras-team/keras

So we could check if they are interested to maintain your PR there.

**Comment by seanpmorgan** [comment]
Thank you for your contribution. We sincerely apologize for any delay in reviewing, but TensorFlow Addons is transitioning to a minimal maintenance and release mode. New features will not be added to this repository. For more information, please see our public messaging on this decision:
[TensorFlow Addons Wind Down](https://github.com/tensorflow/addons/issues/2807)

Please consider sending feature requests / contributions to other repositories in the TF community with a similar charters to TFA:
[Keras](https://github.com/keras-team/keras)
[Keras-CV](https://github.com/keras-team/keras-cv)
[Keras-NLP](https://github.com/keras-team/keras-nlp)
